ndev/services/packer.py

```diff
@@ -424,7 +425,15 @@ def copy_repo_sources(self):
                 message=f"Copying repo source {copy_item.origin} to {copy_item.destination}.",
                 verbosity=Verbosity.VERBOSE.value,
             )
-            repo_url, repo_ref = copy_item.origin, copy_item.ref
+
+            repo_url, repo_ref, package_name = copy_item.origin, copy_item.ref, copy_item.package_name
+            if package_name:
+                package_name_dep = package_name.replace('_', '-')
+                requirement_line = next((l for l in self._get_requirements_txt_list() if f"{package_name_dep}==" in l))
+                requirement_spec = requirement_line.split(";")[0].strip()
+                package_version = requirement_spec.split("==")[1]
+                repo_ref = f"{package_name}-{package_version}"
```

Comment:
repo_ref can be a string with format "v$VERSION$"

Reply:
fixed